commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -208,4 +220,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Builds URL with image coords for MediaWiki API calls
+     * Example URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories|coordinates|pageprops&format=json&clshow=!hidden&coprop=type|name|dim|country|region|globe&codistancefrompoint=38.11386944444445|13.356263888888888&
+     * generator=geosearch&redirects=&ggscoord=38.11386944444445|13.356263888888888&ggsradius=100&ggslimit=10&ggsnamespace=6&ggsprop=type|name|dim|country|region|globe&ggsprimary=all&formatversion=2
+     */
+    public static class UrlBuilder {
+        private static String buildUrl (String coords){
+
+            Uri.Builder builder = Uri.parse("https://commons.wikimedia.org/").buildUpon();
+
+            builder.appendPath("w")
+                    .appendPath("api.php")
+                    .appendQueryParameter("action", "query")
+                    .appendQueryParameter("prop", "categories|coordinates|pageprops")
+                    .appendQueryParameter("format", "json")
+                    .appendQueryParameter("clshow", "!hidden")
+                    .appendQueryParameter("coprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("codistancefrompoint", coords)
+                    .appendQueryParameter("generator", "geosearch")
+                    .appendQueryParameter("ggscoord", coords)
+                    .appendQueryParameter("ggsradius", "100")
+                    .appendQueryParameter("ggslimit", "10")
+                    .appendQueryParameter("ggsnamespace", "6")
+                    .appendQueryParameter("ggsprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("ggsprimary", "all")
+                    .appendQueryParameter("formatversion", "2");
+
+            return builder.toString();
+        }
+    }
+
 }
```

Comment:
Edit: Ah, okay, I figured the latter problem out. My project module SDK had somehow changed to an invalid copy of Maven Android API 22(1), not really sure why it did that. It happens every time I try to alter the pom.xml.